launchpad-to-stargate/tutorial/02-module-migration.md

```diff
@@ -0,0 +1,688 @@
+	cmd := &cobra.Command{
+		Use:   "create-claim [proof]",
+		Short: "Creates a new claim",
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+      		argsProof := string(args[0])
+
+        	clientCtx := client.GetClientContextFromCmd(cmd)
+			clientCtx, err := client.ReadTxCommandFlags(clientCtx, cmd.Flags())
+			if err != nil {
+				return err
+			}
+
+			msg := types.NewMsgCreateClaim(clientCtx.GetFromAddress().String(), string(argsProof))
+			if err := msg.ValidateBasic(); err != nil {
+				return err
+			}
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), msg)
+		},
+	}
+
+	flags.AddTxFlagsToCmd(cmd)
+
+    return cmd
+}
+
+func CmdUpdateClaim() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "update-claim [id] [proof]",
+		Short: "Update a claim",
+		Args:  cobra.ExactArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+            id := args[0]
+      argsProof := string(args[1])
+
+        	clientCtx := client.GetClientContextFromCmd(cmd)
+			clientCtx, err := client.ReadTxCommandFlags(clientCtx, cmd.Flags())
+			if err != nil {
+				return err
+			}
+
+			msg := types.NewMsgUpdateClaim(clientCtx.GetFromAddress().String(), id, string(argsProof))
+			if err := msg.ValidateBasic(); err != nil {
+				return err
+			}
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), msg)
+		},
+	}
+
+	flags.AddTxFlagsToCmd(cmd)
+
+    return cmd
+}
+
+func CmdDeleteClaim() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "delete-claim [id] [proof]",
+		Short: "Delete a claim by id",
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+            id := args[0]
+
+        	clientCtx := client.GetClientContextFromCmd(cmd)
+			clientCtx, err := client.ReadTxCommandFlags(clientCtx, cmd.Flags())
+			if err != nil {
+				return err
+			}
+
+			msg := types.NewMsgDeleteClaim(clientCtx.GetFromAddress().String(), id)
+			if err := msg.ValidateBasic(); err != nil {
+				return err
+			}
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), msg)
+		},
+	}
+
+	flags.AddTxFlagsToCmd(cmd)
+
+    return cmd
+}
+```
+
+If we compare the two files, we will notice that there is a difference in how the commands are defined. For instance, the Stargate version of the functions no longer require a codec as an argument - eg. `func GetCmdXClaim(cdc *codec.Codec)` vs `func CmdXClaim()`. In this case, we can remove `cdc *codec.Codec` from each of the functions as well as the `"github.com/cosmos/cosmos-sdk/codec"` import.
+
+Another difference you will notice is how the application reads CLI commands. This includes reading the command, building the message, and Generating or broadcasting the command. For instance, the contents of the `GetCmdCreateClaim` command will be updated as such when following the updates generated by Starport and described in further detail [here](https://docs.cosmos.network/master/migrations/app_and_modules.html#cli).
+
```

Comment:
It looks to me that this is the core logic is hashing the input and using the sha256 hash as argsProof, instead of just the string.

Maybe you can add a comment like

```
The thing we can point our attention to, is creating a hash of the input, which is the core logic of our implementation. We have the following code which creates a sha256 hash of our input and saves this with a blockchain transaction that the user can send to our blockchain.
We use this hash as input to `NewMsgCreateClaim ` in our `GetCmdCreateClaim` function.
```

```
hasher := sha256.New()
s, _ := ioutil.ReadFile(args[0])
hasher.Write(s)
argsProof := hex.EncodeToString(hasher.Sum(nil))
```